lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1736,10 +1743,14 @@ class MongoClientInterface {
                 };
                 log.error(
                     'internalListObjectV1: error listing objects', logObj);
+                // call explicitly the destroy method to clean the mongodb cursor
+                stream.destroy();
```

Comment:
is this call not duplicated: e.g. will `end` not be called on `error` as well?

Reply:
No, and actually it's not even called in the v1 case due to the piping. See https://github.com/scality/Arsenal/pull/2177/commits/4aa8b5cc6e82d4f69773e5d3cedcc60b9a22f39c

if I remove the line, the test would fail because the stream was not properly cleaned.